packages/block-editor-utils/src/controls/Number.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { TextControl as NumberControl } from '@wordpress/components';
+import { ControlProps } from '../types/index.js';
+
+function Number<T extends Record<string, any>>({
```

Comment:
Do not use `Number` because it shadows the [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) class. Use `NumberField` instead.